File: src/B_5532/B_5532.java
package B_5532;

import java.io.*;

public class B_5532 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int L = Integer.parseInt(br.readLine());
        int A = Integer.parseInt(br.readLine());
        int B = Integer.parseInt(br.readLine());
        int C = Integer.parseInt(br.readLine());
        int D = Integer.parseInt(br.readLine());
        int s = (A/C) + (A%C==0 ? 0 : 1);
        System.out.println(s);
        int d = (B/D) + (B%D==0 ? 0 : 1);
        System.out.println(d);
        System.out.println(L-Math.max(s,d));
    }
}

File: src/B_5532/B_5532.kt
package B_5532

import java.io.*
import kotlin.math.*

fun main() = with(BufferedReader(InputStreamReader(System.`in`))){
    val L = readLine().toInt()
    val A: Int = readLine().toInt()
    val B = readLine().toInt()
    val C: Int = readLine().toInt()
    val D = readLine().toInt()
    val s = A / C + if (A % C == 0) 0 else 1
    val d = B / D + if (B % D == 0) 0 else 1
    print(L- max(s,d))
}